pom.xml

```diff
@@ -210,6 +211,82 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>glassfish</id>
+      <properties>
+        <!-- Application properties -->
+        <webapp.graphTraversalUrl>http://localhost:8080/cargo-tracker/rest/graph-traversal/shortest-path</webapp.graphTraversalUrl>
+        <db.driverClass>org.h2.jdbcx.JdbcDataSource</db.driverClass>
+        <!-- H2 file engine requires an absolute path, eg. ~/name, ./name or a full-qualified
+          path name (it should include driver name under Windows). -->
+        <db.jdbcUrl>jdbc:h2:file:./cargo-tracker-data/cargo-tracker-database</db.jdbcUrl>
+        <db.user />
+        <db.password />
+      </properties>
+      <dependencies>
+        <!-- We need a Jersey specific dependency for property name constants -->
+        <dependency>
+          <groupId>org.glassfish.jersey.core</groupId>
+          <artifactId>jersey-server</artifactId>
+          <version>3.1.2</version>
+          <scope>provided</scope>
+        </dependency>
+        <!-- Embedding the JDBC driver in the WAR works in GlassFish -->
+        <dependency>
+          <groupId>com.h2database</groupId>
+          <artifactId>h2</artifactId>
+          <version>2.1.214</version>
+          <scope>runtime</scope>
+        </dependency>
+
+        <!-- The Arquillian connector for GlassFish. This lets Arquillian
+          start and stop GlassFish, and deploy and undeploy war files to it. -->
+        <dependency>
+          <groupId>org.omnifaces.arquillian</groupId>
```

Comment:
I'm curious to understand why you've chosen to use a fork of the Arquillian project's code rather than just use the connector produced by the Arquillian project. What does this artifact offer that the upstream does not?

Reply:
Hi @waynebeaton , @m-reza-rahman , there are a few reasons I didn't use the connector from the Arquillian project.

* The connector from the Arquillian project is still a work in progress. The latest release in Maven Central is 1.0.0.Alpha1, it depends on a milestone artifact, which isn't in Maven Central. The name of the artifact in the master branch in https://github.com/arquillian/arquillian-container-glassfish6 has changed since the `1.0.0.Alpha1` was released.
* There's no support for the GlassFish Embedded connector yet. This is clearly stated in the code repository, and there's no hint that support for it will be added soon. The GlassFish Embedded connector starts faster, it's easier to debug, and is much easier to configure in the project.
* There's no official documentation for the connector from the Arquillian project. Currently not even for the older versions, it's lost, all links to the old documentation don't work anymore. The only documentation is the repository with the source code and this website from Hantsy: https://hantsy.github.io/jakartaee9-starter-boilerplate/arq-glassfish.html

Compared to that, the connector by OmniFish (our company), is mature, actively maintained, supports GlassFish Embedded and even supports more features. It's based on the code from the Arquillian project, but this source code was modified by Payara and then modified by OmniFish. So it's not easy to donate it back to the Arquillian project. Payara would have to contribute their changes first. We would like to move it to the Eclipse Foundation, but that would even require a donation from the Arquillian project.

We try to be as helpful to the GlassFish community as possible. With the OmniFish Arquillian connector for GlassFish, we aim to provide the best Arquillian connector possible without any limitations. But it's sometimes hard to contribute back to the original projects like Arquillian or GlassFish because of legal measures or because other parties are not collaborative enough. If you have any idea how to support the community better, please, let us know.

Reply:
I got the GlassFish profile working with the GlassFish Managed connector from the Arquillian project here: https://github.com/eclipse-ee4j/cargotracker/pull/282. However, I feel it's a suboptimal solution because of all the drawbacks with that connector I mentioned, so I only raised it as a draft PR for you to see how it would look like.

Reply:
I suggest we work to merge #282. In the meanwhile, I suggest kindly closing this PR. I hope the OmniFish enhancements can be donated either to the Arquillian project or the Eclipse Foundation. I am sure the Foundation is willing to help.

Reply:
OK, we can use #282 PR then. 

However, the OmniFish enhancements are on top of Payara's enhancements, and we can't contribute code with Payara copyright to Eclipse foundation. But we might be able to contribute files with Payara copyright to Arquillian, which allows to keep the copyright by the original authors.

Reply:
I think contributing back to Arquillian would suit this project well enough and perhaps even the Foundation is OK with that?